distribution/pull_v2.go

```diff
@@ -94,12 +95,24 @@ func (p *v2Puller) Pull(ctx context.Context, ref reference.Named) (err error) {
 
 func (p *v2Puller) checkTrusted(ref reference.Named) (reference.Named, error) {
 	p.originalRef = ref
-	imgRef, err := docker.NewReference(ref)
+	dockerRef, err := containersImageRef.ParseNamed(ref.String())
```

Comment:
@mtrmac PTAL at this since in projectatomic/docker it's the only way to have a consistent reference parsing for signatures. It's working like this, pinging you to understand if I'm missing something. (this is related to https://github.com/containers/image/pull/123)